packages/dds/tree/src/simple-tree/objectNode.ts

```diff
@@ -502,3 +509,29 @@ function assertUniqueKeys<
 		derivedStoredKeys.add(storedKey);
 	}
 }
+
+/**
+ * Creates a policy for allowing unknown optional fields on an object node which delegates to the policy defined
+ * on the object node's internal schema data.
+ */
+export function createUnknownOptionalFieldPolicy(
+	schema: ImplicitFieldSchema,
+): SchemaPolicy["allowUnknownOptionalFields"] {
+	const identifierToObjectSchema = new Map<
+		string,
+		ObjectNodeSchema & ObjectNodeSchemaInternalData
+	>();
+
+	walkFieldSchema(schema, {
```

Comment:
we technically already do the algorithmically expensive bit (walking the schema) and could leverage `getUnhydratedContext` if we wanted. However, this would also mean plumbing the view schema and not just the stored schema into the validation functions in `schemaChecker.ts`, which IMO makes the signature kind of ugly whereas including this as a degree of freedom in the 'schema policy' feels more flexible / easy to unit test / etc. So I went with this sort of approach, figuring it's an implementation detail anyway.

Reply:
I realized we can have best of both worlds--inject the policy, but leverage getUnhydratedContext in the implementation here. Doing that in update before merging.